base/show.jl

```diff
@@ -2559,14 +2559,12 @@ function show_tuple_as_call(out::IO, name::Symbol, sig::Type;
     print_within_stacktrace(io, ")", bold=true)
     show_method_params(io, tv)
     str = String(take!(unwrapcontext(io)[1]))
-    if get(out, :limit, false)::Bool
+    typelimitflag = get(out, :stacktrace_types_limited, nothing)
+    if typelimitflag !== nothing
```

Comment:
```suggestion
    if typelimitflag isa RefValue{Bool}
```
for better inference?